File: LeetCode/Easy/2138-Divide_A_String_Into_Groups_Of_Size_K/2138-DivideAStringIntoGroupsOfSizeK.cpp
class Solution {
public:
    vector<string> divideString(string s, int k, char fill) {
        int n = s.size();

        if (n % k != 0) {
            s.append((k - (n % k)), fill);
        }

        std::vector<std::string> ans;

        for (int i = 0; i < n; i+=k) {
            ans.push_back(s.substr(i, k));
        }

        return ans;
    }
};

File: LeetCode/Easy/2138-Divide_A_String_Into_Groups_Of_Size_K/2138-DivideAStringIntoGroupsOfSizeK.js
/**
 * @param {string} s
 * @param {number} k
 * @param {character} fill
 * @return {string[]}
 */
var divideString = function (s, k, fill) {
    let n = s.length;

    if (n % k != 0) {
        s = s + fill.repeat(k - (n % k));
    }

    let ans = [];

    for (let i = 0; i < s.length; i += k) {
        ans.push(s.substring(i, i + k));
    }

    return ans
};